exercises/largest-series-product/src/example.c

```diff
@@ -0,0 +1,54 @@
+#include <string.h>
+#include <stdbool.h>
+#include "largest_series_product.h"
+
+#define MAX(x, y) (x) > (y) ? (x) : (y)
+
+static int to_int(char c)
+{
+   return c - '0';
+}
+
+static bool is_digit(char c)
+{
+   return (c >= '0') && (c <= '9');
+}
+
+long largest_series_product(char *digits, size_t span)
+{
+   size_t digit_count = strlen(digits);
+   size_t zeros = 0;
+   long product = 1;
+   long largest_product = 0;
+
+   if (span > digit_count)
+      return -1;
+
+   if (!span && !digit_count)
+      return 1;
+
+   for (size_t i = 0; i < digit_count; i++) {
+      if (!is_digit(digits[i]))
+         return -1;
+
+      int to_add = to_int(digits[i]);
```

Comment:
There are two reasons that I opted to go this route:
- `'0'` also returns `0` so you can't use this alone to determine whether an invalid character was provided.
- `atoi` wants a null-terminated string and not a single character. This means we'd have to construct a null-terminated string on the fly in order to use it.